ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracingshim/__init__.py

```diff
@@ -0,0 +1,257 @@
+
+        # Return self for call chaining.
+        return self
+
+    def finish(self, finish_time=None):
+        end_time = finish_time
+        if end_time is not None:
+            end_time = util.time_seconds_to_ns(finish_time)
+        self._otel_span.end(end_time=end_time)
+
+    def set_tag(self, key, value):
+        self._otel_span.set_attribute(key, value)
+
+        # Return self for call chaining.
+        return self
+
+    def log_kv(self, key_values, timestamp=None):
+        if timestamp is not None:
+            event_timestamp = util.time_seconds_to_ns(timestamp)
+        else:
+            event_timestamp = None
+
+        event_name = util.event_name_from_kv(key_values)
+        self._otel_span.add_event(event_name, event_timestamp, key_values)
+
+        # Return self for call chaining.
+        return self
+
+    def set_baggage_item(self, key, value):
+        logger.warning(
+            "Calling unimplemented method set_baggage_item() on class %s",
+            self.__class__.__name__,
+        )
+        # TODO: Implement.
+
+    def get_baggage_item(self, key):
+        logger.warning(
+            "Calling unimplemented method get_baggage_item() on class %s",
+            self.__class__.__name__,
+        )
+        # TODO: Implement.
+
+    # TODO: Verify calls to deprecated methods `log_event()` and `log()` on
+    # base class work properly (it's probably fine because both methods call
+    # `log_kv()`).
+
+
+class ScopeWrapper(opentracing.Scope):
+    def close(self):
+        self._span.finish()
+        # TODO: Set active span on OpenTelemetry tracer.
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+
+class ScopeManagerWrapper(opentracing.ScopeManager):
+    def __init__(self, tracer):
+        # The only thing the `__init__()` method on the base class does is
+        # initialize `self._noop_span` and `self._noop_scope` with no-op
+        # objects. Therefore, it doesn't seem useful to call it.
+        # pylint: disable=super-init-not-called
+        self._tracer = tracer
+
+    @contextmanager
+    def activate(self, span, finish_on_close):
+        with self._tracer.unwrap().use_span(
+            span.unwrap(), end_on_exit=finish_on_close
+        ) as otel_span:
+            wrapped_span = SpanWrapper(
+                self._tracer, otel_span.get_context(), otel_span
+            )
+            yield ScopeWrapper(self, wrapped_span)
+
+    @property
+    def active(self):
+        span = self._tracer.unwrap().get_current_span()
+        if span is None:
+            return None
+
+        wrapped_span = SpanWrapper(self._tracer, span.get_context(), span)
+        return ScopeWrapper(self, wrapped_span)
+        # TODO: Return a saved instance of SpanWrapper instead of constructing
+        # a new object (and the same for ScopeWrapper?).
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
```

Comment:
I like Christian's idea (and this is something we will similarly use for supporting Baggage as well ;) )

Reply:
(But that can be done in a separate PR, as an improvement)

Reply:
OK. I'm resolving this for now and leaving the TODO in the code. We can revisit this at a later stage.